src/viper/trans.ml

```diff
@@ -30,8 +30,30 @@ let rec unit (u : Mo_def.Syntax.comp_unit) : prog =
   match body.it with
   | M.ActorU(id_opt, decs) ->
      let ctxt = { self = None; ids = Env.empty } in
-     let (ctxt', mk_is) = dec_fields ctxt decs in
-     let is = List.map (fun mk_i -> mk_i ctxt') mk_is in
+     let ctxt', inits, mk_is = dec_fields ctxt decs in
+     let is' = List.map (fun mk_i -> mk_i ctxt') mk_is in
+
+     let init_id = {it = "__init__"; at = Source.no_region; note = NoInfo } in
+     let self_id = {it = "$Self"; at = Source.no_region; note = NoInfo } in
```

Comment:
isn't `id` the helper for this? And then inline the two values?

Reply:
Okay, scratch that, let's intro an `id_at` for this, so that we can supply a region, use from line 98 too.